x-pack/plugin/esql/src/main/antlr/EsqlBaseParser.g4

```diff
@@ -246,9 +246,13 @@ showCommand
     ;
 
 enrichCommand
-    : ENRICH policyName=fromIdentifier (ON matchField=qualifiedNamePattern)? (WITH enrichWithClause (COMMA enrichWithClause)*)?
+    : ENRICH (enrichMode=enrichPolicyMode)? policyName=fromIdentifier (ON matchField=qualifiedNamePattern)? (WITH enrichWithClause (COMMA enrichWithClause)*)?
```

Comment:
Should the `mode` be after the policy name? I am fine with either option.

Reply:
That's the first option I went for however I wanted to be consistent with FROM namely have the options for a command specified after its declaration not per parameter, e.g.:

```
ENRICH policy[mode=any]   // it's clear mode affects the policy however is fragile as it could be interpreted as being
							// part of the policy name. Right now the index doesn't include[] but 
							// if this gets changed, the declaration will break
ENRICH "policy[with brackets]"[mode=any]   // not as pretty
ENRICH [mode= any] "policyWithBracket[]" 	// longer but safer to evolve
```

Another idea is to namespace the parameter to indicate what this mode is about since not everyone would be using CCQ and use something like ccq.mode, ccq.policy, ccq.policy.mode, ccq.policy.resolution:

`ENRICH [ccq.policy.mode=any] policyName ON field WITH ...`

Reply:
Thanks for explaining. I am fine with either option.

Reply:
`ENRICH policy`
`ENRICH FROM ANY policy`
`ENRICH FROM COORDINATING policy`
`ENRICH FROM REMOTE policy`
?

Reply:
I like @nik9000's suggestion and I am kind of torn between the two approaches:
- one is consistent with `FROM index [metadata _source]` approach where we put stuff between square brackets in the form of settings, or additional instructions. Whenever we have a similar setting in the future for commands we should probably follow the same style.
- `ENRICH` is kind of special because the command itself is a series of natural language words put together to form a sentence. For example `ENRICH languages_policy ON language WITH name = language_name` would become:
```
ENRICH FROM COORDINATING languages_policy ON language WITH name = language_name
```
the drawback, though, is that if we add any additional elements to ENRICH in the future we need to do it the same natural language way, whereas with a square brackets approach the mechanism allows unlimited extensibility (just a made up setting name): `[policy.mode=coordinating, node.affinity=hot]`

Now, that I listed both options, I tend to favor the consistent square brackets option.

Reply:
FWIW, I'd also find it more fluent keeping the consistency with the `ENRICH` command itself (rather than `FROM` command) and add a `FROM` clause after the policy name. 
OTOH, `FROM` command is an outlier to receive parameters in square brackets, so these one here would bring some balance (but I'd still like a query language with fewer `[`, `]`).

Reply:
FROM flows better however it's not a full property of ENRICH since it only makes sense when running a query in a CCQ environment against a CCQ index.
1. FROM would have to apply only to the policy (which is not an index)
2. without a remote index, command: `ENRICH FROM COORDINATING language` is not only meaningless but incorrect
3. we're hoping the setting is an advance one - namely by default folks would use the `any` mode.

For these reasons I think we should give it less importance than a regular property.

Essentially this introduces a slightly separate mode:
`FROM [medata _index, _source]`
`ENRICH [ccq.mode=any]`

metadata does NOT use `=` and has a list of values
ccq.mode uses `=` and has only one value.

We should align the two so either:
`FROM [metadata=_index,_source]` OR `ENRICH [ccq.mode any]`
In both cases, we need to figure out how to handle enumeration of other options that might be added in the future.
A quick proposal is to add another square bracket
`ENRICH [ccq.mode any] [another.param value], etc...`
A bit more verbose but easier to navigate.

Reply:
> without a remote index, command: ENRICH FROM COORDINATING language is not only meaningless but incorrect

I guess `ENRICH [ccq.mode=remote]` would still err if run locally. However, if we want this to be an "advanced" setting, it might make sense to use `[`, `]` then.

> `OR ENRICH [ccq.mode any]`

👍 

> `ENRICH [ccq.mode any] [another.param value], etc...`

👍

Reply:
> I guess ENRICH [ccq.mode=remote] would still err if run locally. However, if we want this to be an "advanced" setting, it might make sense to use [, ] then.

We haven't decided yet if we want to lenient or not:
1. declaring any ccq property without a ccq index could be considered an error to force the user to remove the setting
2. instead of throwing an error, we could just throw a warning and ignore the setting. this way the same setting can be
used across difference environments; the downside though is the user might end up ignoring the warning.

Reply:
Three options should work whether remote indices are present or not.

- `ANY` means that enrich takes place on any cluster.
- `COORDINATING` means that enrich takes place on the coordinating cluster receiving an ESQL.
- `REMOTE` means that enrich takes place on the cluster hosting the target index.

For local indices, all options have the same semantics. These options affect only the remote indices.